WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -213,6 +213,12 @@
         sign_in organizer
       end
 
+      it 'cannot confirm competition' do
+        patch :update, id: competition, competition: { name: competition.name }, commit: "Confirm"
```

Comment:
Depending on the layout you could either add another form just for the confirmation button or use the [`formaction`](http://devdocs.io/html/element/button) attribute to overwrite the `form[action]` URL. I've never used the latter option and am not sure about browser compatibility, though.